src/sdformat_urdf.cpp

```diff
@@ -0,0 +1,462 @@
+    if (!sdf_collision) {
+      errors.emplace_back(
+        sdf::ErrorCode::STRING_READ,
+        "Failed to get collision on lilnk [" + sdf_link.Name() + "]");
+      return nullptr;
+    }
+
+    auto urdf_collision = std::make_shared<urdf::Collision>();
+
+    urdf_collision->name = sdf_collision->Name();
+
+    // URDF collision is relative to link origin
+    ignition::math::Pose3d collision_pose;
+    sdf::Errors pose_errors = sdf_collision->SemanticPose().Resolve(collision_pose, sdf_link.Name());
+    if (!pose_errors.empty()) {
+      errors.insert(errors.end(), pose_errors.begin(), pose_errors.end());
+      errors.emplace_back(
+        sdf::ErrorCode::STRING_READ,
+        "Failed to get transfrom from collision [" + sdf_collision->Name()
+        + "] to link [" + sdf_link.Name() + "]");
+      return nullptr;
+    }
+    urdf_collision->origin = convert_pose(collision_pose);
+
+    urdf_collision->geometry = convert_geometry(*sdf_collision->Geom(), errors);
+    if (!urdf_collision->geometry) {
+      errors.emplace_back(
+        sdf::ErrorCode::STRING_READ,
+        "Failed to convert geometry on collision [" + sdf_collision->Name()+ "]");
+      return nullptr;
+    }
+
+    if (0u == vi) {
+      urdf_link->collision = urdf_collision;
+    }
+    urdf_link->collision_array.push_back(urdf_collision);
+  }
+
+  return urdf_link;
+}
+
+urdf::JointSharedPtr
+sdformat_urdf::convert_joint(const sdf::Joint & sdf_joint, sdf::Errors & errors)
+{
+  urdf::JointSharedPtr urdf_joint = std::make_shared<urdf::Joint>();
+
+  urdf_joint->name = sdf_joint.Name();
+
+  switch (sdf_joint.Type()) {
+    case sdf::JointType::CONTINUOUS:
+      urdf_joint->type = urdf::Joint::CONTINUOUS;
+      break;
+    case sdf::JointType::REVOLUTE:
+      urdf_joint->type = urdf::Joint::REVOLUTE;
+      break;
+    case sdf::JointType::FIXED:
+      urdf_joint->type = urdf::Joint::FIXED;
+      break;
+    case sdf::JointType::PRISMATIC:
+      urdf_joint->type = urdf::Joint::PRISMATIC;
+      break;
+    case sdf::JointType::INVALID:     // Unsupported: fall through to default
+    case sdf::JointType::BALL:        //  |
+    case sdf::JointType::GEARBOX:     //  |
+    case sdf::JointType::REVOLUTE2:   //  |
+    case sdf::JointType::SCREW:       //  |
+    case sdf::JointType::UNIVERSAL:   //  V
+    default:
+      errors.emplace_back(
+        sdf::ErrorCode::STRING_READ,
+        "Unsupported joint type on joint [" + sdf_joint.Name() + "]");
+      return nullptr;
+  };
+
+  const sdf::JointAxis * sdf_axis = sdf_joint.Axis(0);
+  urdf_joint->axis.x = sdf_axis->Xyz().X();
+  urdf_joint->axis.y = sdf_axis->Xyz().Y();
+  urdf_joint->axis.z = sdf_axis->Xyz().Z();
+
+  urdf_joint->child_link_name = sdf_joint.ChildLinkName();
+  urdf_joint->parent_link_name = sdf_joint.ParentLinkName();
+
+  return urdf_joint;
+}
+
+urdf::Pose
+sdformat_urdf::convert_pose(const ignition::math::Pose3d & sdf_pose)
```

Comment:
Nit: `#include <ignition/math/Pose3.hh`

Reply:
Included header in ab6637bcb03d00f08facd94da7956ffe25cb6bf4